packages/dcos-integration-test/extra/test_users.py

```diff
@@ -0,0 +1,162 @@
+
+Assume that access control is activated in Master Admin Router (could be
+disabled with `oauth_enabled`) and therefore authenticate individual HTTP
+requests.
+
+One aspect of DC/OS user management is that once authenticated a user can add
+other users. Unauthenticated HTTP requests are rejected by Master Admin Router
+and user management fails (this is the coarse-grained authorization model of
+(open) DC/OS). Here, test that unauthenticated HTTP requests cannot manage
+users. However, do not test that newly added users can add other users: in this
+test suite we are limited to having authentication state for just a single user
+available. This is why we can test managing other users only from that first
+user's point of view. That is, we can not test that a user (e.g. user2) which
+was added by the first user (user1) can add another user (user3).
+"""
+import logging
+
+import pytest
+
+from test_helpers import expanded_config
+
+
+__maintainer__ = 'jgehrcke'
+__contact__ = 'security-team@mesosphere.io'
+
+
+log = logging.getLogger(__name__)
+
+
+# Skip entire module in downstream integration tests.
+if 'security' in expanded_config:
+    pytest.skip(
+        'Skip upstream-specific user management tests',
+        allow_module_level=True
+    )
+
+
+def get_users(apisession):
+    r = apisession.get('/acs/api/v1/users')
+    r.raise_for_status()
+    users = {u['uid']: u for u in r.json()['array']}
+    return users
+
+
+def delete_user(apisession, uid):
+    r = apisession.delete('/acs/api/v1/users/%s' % (uid, ))
+    r.raise_for_status()
+    assert r.status_code == 204
+
+
+@pytest.fixture()
+def remove_users_added_by_test(dcos_api_session):
+    users_before = set(get_users(dcos_api_session))
+    log.info('remove_users_added_by_test pre test: users are %s', users_before)
+    try:
+        yield
+    finally:
+        users_after = set(get_users(dcos_api_session))
+        new_uids = users_after - users_before
+        for uid in new_uids:
+            log.info('remove_users_added_by_test post test: remove `%s`', uid)
+            delete_user(dcos_api_session, uid)
+
+
+def test_users_get(dcos_api_session):
+    users = get_users(dcos_api_session)
+    assert users
+
+    required_keys = ('uid', 'description')
+    for _, userdict in users.items():
+        for k in required_keys:
+            assert k in userdict
+
+
+def test_user_put_no_email_uid(dcos_api_session):
+    r = dcos_api_session.put('/acs/api/v1/users/user1')
+
+    # This is the current behavior. It does not need to stay in future versions.
+    assert r.status_code == 500
+    assert 'invalid email' in r.text
+
+
+@pytest.mark.usefixtures('remove_users_added_by_test')
+def test_user_put_email_uid(dcos_api_session):
+    # The current behavior is that the request body can be an empty JSON
+    # document. It does not need to stay in future versions.
+    r = dcos_api_session.put('/acs/api/v1/users/user1@email.de', json={})
```

Comment:
The `email.de` is an actual domain with MX record. Do we want to use it in tests?

Reply:
Ha, good point. Well :-) I think in one of the next iterations I will indeed get rid of that again.